### ISSUE TITLE
Could not find Cargo.toml in hidden directories

### ISSUE THREAD

**Issue by lilydjwg** [rust-lang/cargo#1999]
```bash
>>> cargo new --bin test
>>> mv test .test
>>> cd .test
>>> cargo build
Unable to update file:///home/lilydjwg/tmpfs/.test

Caused by:
  Could not find Cargo.toml in `/home/lilydjwg/tmpfs/.test`
```

**Comment by alexcrichton** [comment]
I believe this is an instance of [this check](https://github.com/rust-lang/cargo/blob/c76e9d13db3fac1f234777c462469cb2bc49ee17/src/cargo/ops/cargo_read_manifest.rs#L46-L50) being a little overeager, and should probably just have the `dir != path` condition like below to ensure that you can recursive if the top-level is a dot-directory but you don't recurse further.

Thanks for the report1

**Comment by Kintaro** [comment]
Can confirm. That destroyed my window manager's config directory.

**Comment by phi-gamma** [comment]
Hey guys, I’ve submitted a PR implementing @alexchrichton’s suggestion
that fixes the wtftw config build for me.

**Comment by alexcrichton** [comment]
Fixed in #2019
